File: app/src/main/java/com/code_design_camp/client/friday/HeadDisplayClient/fragments/store/ErrorFragment.java
package com.code_design_camp.client.friday.HeadDisplayClient.fragments.templateClasses;


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.fragment.app.Fragment;

import com.code_design_camp.client.friday.HeadDisplayClient.R;
import com.code_design_camp.client.friday.HeadDisplayClient.fragments.store.MainStoreFragment;
import com.google.firebase.firestore.FirebaseFirestoreException;


/**
 * A simple {@link Fragment} subclass.
 */
public class ErrorFragment extends Fragment {
    private FirebaseFirestoreException e;


    public ErrorFragment(FirebaseFirestoreException e) {
        this.e = e;
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View v = inflater.inflate(R.layout.fragment_error, container, false);
        Button retry = v.findViewById(R.id.retry);
        ((TextView) v.findViewById(R.id.errTitle)).setText(getString(R.string.loading_err_title,String.valueOf(e.getCode().value())));
        ((TextView) v.findViewById(R.id.subtitle_err)).setText(e.getLocalizedMessage());
        retry.setOnClickListener((view) -> getFragmentManager().beginTransaction()
                .add(R.id.store_frag_container,new MainStoreFragment())
                .remove(this)
                .commit());
        return v;
    }

}

File: appModules/store/src/main/java/com/friday/ar/store/ui/fragments/ErrorFragment.kt
package com.friday.ar.store.ui.fragments


import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.friday.ar.core.util.firebase.FireStoreCodeInterpreter
import com.friday.ar.store.R
import com.google.firebase.firestore.FirebaseFirestoreException
import kotlinx.android.synthetic.main.fragment_error.*


/**
 * Error Fragment which shows an error page to the user.
 *
 * @see MainStoreFragment, com.friday.HeadDisplayClient.store.fragments.StoreFeaturedFragment
 */
class ErrorFragment(private val e: FirebaseFirestoreException) : Fragment() {
    private lateinit var mContext: Context

    override fun onAttach(context: Context) {
        super.onAttach(context)
        this.mContext = context
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_error, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        val errInterpreter = FireStoreCodeInterpreter(mContext, e)
        super.onViewCreated(view, savedInstanceState)
        errTitle.text = getString(R.string.loading_err_title, errInterpreter.code.toString())
        subtitle_err.text = errInterpreter.message
        fullMessage.text = errInterpreter.exceptionMessage
    }

}
